### PR TITLE
rust: update `cmake` dependency ranges

### PR DESCRIPTION
**Pull request by simonLeary42**
fixes the cmake dependency versions for the rust package.

To find this information, I cloned the Rust fork of the LLVM project, and grepped the Cmake files after checking out each Rust release tag:
```
$ git tag | grep rustc | sort -n | while read -r tag; do printf "$tag -> "; git checkout -q "$tag"; grep cmake_minimum_required ./lld/CmakeLists.txt | sed -E 's|^.*VERSION ([^\)]+).*$|\1|'; done
rustc-1.34.0 -> 3.4.3
rustc-1.34.1 -> 3.4.3
rustc-1.34.2 -> 3.4.3
rustc-1.35.0 -> 3.4.3
rustc-1.36.0 -> 3.4.3
rustc-1.37.0 -> 3.4.3
rustc-1.38.0 -> 3.4.3
rustc-1.39.0 -> 3.4.3
rustc-1.40.0 -> 3.4.3
rustc-1.41.0 -> 3.4.3
rustc-1.41.1 -> 3.4.3
rustc-1.42.0 -> 3.4.3
rustc-1.43.0 -> 3.4.3
rustc-1.43.1 -> 3.4.3
rustc-1.44.0 -> 3.4.3
rustc-1.44.1 -> 3.4.3
rustc-1.45.0 -> 3.4.3
rustc-1.45.1 -> 3.4.3
rustc-1.45.2 -> 3.4.3
rustc-1.46.0 -> 3.4.3
rustc-1.47.0 -> 3.4.3
rustc-1.48.0 -> 3.4.3
rustc-1.49.0 -> 3.4.3
rustc-1.50.0 -> 3.4.3
rustc-1.51.0 -> 3.4.3
rustc-1.52.0 -> 3.13.4
rustc-1.52.1 -> 3.13.4
rustc-1.53.0 -> 3.13.4
rustc-1.54.0 -> 3.13.4
rustc-1.55.0 -> 3.13.4
rustc-1.56.0 -> 3.13.4
rustc-1.56.1 -> 3.13.4
rustc-1.57.0 -> 3.13.4
rustc-1.58.0 -> 3.13.4
rustc-1.58.1 -> 3.13.4
rustc-1.59.0 -> 3.13.4
rustc-1.60.0 -> 3.13.4
rustc-1.61.0 -> 3.13.4
rustc-1.62.0 -> 3.13.4
rustc-1.62.1 -> 3.13.4
rustc-1.63.0 -> 3.13.4
rustc-1.64.0 -> 3.13.4
rustc-1.65.0 -> 3.13.4
rustc-1.66.0 -> 3.13.4
rustc-1.66.1 -> 3.13.4
rustc-1.67.0 -> 3.13.4
rustc-1.67.1 -> 3.13.4
rustc-1.68.0 -> 3.13.4
rustc-1.68.1 -> 3.13.4
rustc-1.68.2 -> 3.13.4
rustc-1.69.0 -> 3.13.4
rustc-1.70.0 -> 3.13.4
rustc-1.71.0 -> 3.13.4
rustc-1.71.1 -> 3.13.4
rustc-1.72.0 -> 3.13.4
rustc-1.72.1 -> 3.13.4
rustc-1.73.0 -> 3.20.0
rustc-1.74.0 -> 3.20.0
rustc-1.74.1 -> 3.20.0
rustc-1.75.0 -> 3.20.0
rustc-1.76.0 -> 3.20.0
rustc-1.77.0 -> 3.20.0
rustc-1.77.1 -> 3.20.0
rustc-1.77.2 -> 3.20.0
rustc-1.78.0 -> 3.20.0
rustc-1.79.0 -> 3.20.0
```

### CODE REVIEW

**Comment by alecbcs** [comment]
@spackbot fix style

**Review by bernhardkaindl — APPROVED** [review]

**Comment by alecbcs** [comment]
@bernhardkaindl please don't merge PRs in the future if someone from the core team is assigned to them. I had feedback for this PR and didn't want to merge as is.

**Comment by bernhardkaindl** [comment]
I had hoped that one may set it to draft, or a comment would be made public if there was something new to know.

@alecbcs, thanks for the explanation about the assignment from the core team!

**Pull request merged**
